tasks/src/lib.rs

```diff
@@ -93,108 +87,79 @@ pub struct TaskDescriptor {
     pub prover: String,
 }
 
-impl TaskDescriptor {
-    pub fn to_vec(self) -> Vec<u8> {
-        self.into()
-    }
-}
-
-impl From<TaskDescriptor> for Vec<u8> {
-    fn from(val: TaskDescriptor) -> Self {
-        let mut v = Vec::new();
-        v.extend_from_slice(&val.chain_id.to_be_bytes());
-        v.extend_from_slice(val.blockhash.as_ref());
-        v.extend_from_slice(&(val.proof_system as u8).to_be_bytes());
-        v.extend_from_slice(val.prover.as_bytes());
-        v
-    }
-}
-
-// Taskkey from EnqueueTaskParams
-impl From<&EnqueueTaskParams> for TaskDescriptor {
-    fn from(params: &EnqueueTaskParams) -> TaskDescriptor {
+impl From<(ChainId, B256, ProofType, String)> for TaskDescriptor {
+    fn from(
+        (chain_id, blockhash, proof_system, prover): (ChainId, B256, ProofType, String),
+    ) -> Self {
         TaskDescriptor {
-            chain_id: params.chain_id,
-            blockhash: params.blockhash,
-            proof_system: params.proof_type,
-            prover: params.prover.clone(),
+            chain_id,
+            blockhash,
+            proof_system,
+            prover,
         }
     }
 }
 
-impl From<(ChainId, B256, ProofType, Option<String>)> for TaskDescriptor {
+impl From<TaskDescriptor> for (ChainId, B256) {
     fn from(
-        (chain_id, blockhash, proof_system, prover): (ChainId, B256, ProofType, Option<String>),
-    ) -> Self {
         TaskDescriptor {
             chain_id,
             blockhash,
-            proof_system,
-            prover: prover.unwrap_or_default(),
-        }
+            ..
+        }: TaskDescriptor,
+    ) -> Self {
+        (chain_id, blockhash)
     }
 }
 
-#[derive(Debug, Clone)]
-pub struct TaskProvingStatus(pub TaskStatus, pub Option<String>, pub DateTime<Utc>);
+/// Task status triplet (status, proof, timestamp).
+pub type TaskProvingStatus = (TaskStatus, Option<String>, DateTime<Utc>);
 
 pub type TaskProvingStatusRecords = Vec<TaskProvingStatus>;
 
+pub type TaskReport = (TaskDescriptor, TaskStatus);
+
 #[derive(Debug, Clone)]
 pub struct TaskManagerOpts {
     pub sqlite_file: PathBuf,
     pub max_db_size: usize,
 }
 
-#[derive(Debug, Serialize, ToSchema)]
-pub struct TaskReport(pub TaskDescriptor, pub TaskStatus);
-
 #[async_trait::async_trait]
-pub trait TaskManager {
-    /// new a task manager
+pub trait TaskManager: IdStore + IdWrite {
```

Comment:
Can idstore implicitly cover idwriter? Or maybe better to separate store(read) and write like this?

Reply:
For our current use case it could just be used inside one trait, but if we end up adding multiple interfaces it might be handier to have them separated. I've also seen this practice in larger Rust codebases so it might help us down the line.